src/language-js/embed.js

```diff
@@ -316,10 +317,9 @@ function replacePlaceholders(quasisDoc, expressionDocs) {
         replaceCounter++;
       });
 
-      parts.splice(idx + tokensCount, 1, ...tokens);
-      tokensCount += tokens.length - 1;
+      replacedParts.push(...tokens);
```

Comment:
Why is the `tokens` array needed? Why not push directly to `replacedParts`?

Reply:
Also, what if `placeholderMatch === null`?

Reply:
I think the code can be simplified further if you pass the regex to `split` like it's done in the `printHtmlTemplateLiteral` function in the same file. What do you think?

Reply:
Done.